src/main/java/goblintserver/GoblintClient.java

```diff
@@ -55,15 +60,25 @@ public boolean connectGoblitClient() {
      * Method for sending the requests to Goblint server.
      */
 
-    public void writeMessageToSocket(String message) throws IOException {
-        ByteBuffer buffer = ByteBuffer.allocate(1024);
-        buffer.clear();
-        buffer.put(message.getBytes());
-        buffer.flip();
-        while (buffer.hasRemaining()) {
-            channel.write(buffer);
-        }
-        log.debug("Message written to socket.");
+    public void writeRequestToSocket(String request) throws IOException {
+        OutputStream stream = Channels.newOutputStream(channel);
+        stream.write(request.getBytes());  
+        log.info(request);   
+        log.info("Request written to socket.");
+    }
+
+
+    /**
+     * Method for reading the results from Goblint server.
+     */
+
+    public String readResultFromSocket() throws IOException {
+        InputStream stream = Channels.newInputStream(channel);
+        BufferedReader reader = new BufferedReader(new InputStreamReader(stream));
+        String result = reader.readLine();
```

Comment:
Same with the `InputStream` (and its reader) here. With buffering, these recreated streams might cause some unflushed data to be lost between calls.